client/me/purchases/manage-purchase/change-payment-method/index.jsx

```diff
@@ -157,32 +164,41 @@ ChangePaymentMethod.propTypes = {
 };
 
 const wpcom = wp.undocumented();
-const wpcomGetStoredCards = () => wpcom.getStoredCards();
+const wpcomAssignPaymentMethod = ( subscriptionId, stored_details_id, fn ) =>
+	wpcom.assignPaymentMethod( subscriptionId, stored_details_id, fn );
 
-function ChangePaymentMethodList() {
-	const [ formSubmitting ] = useState( false );
-	const translate = useTranslate();
+function ChangePaymentMethodList( {
+	currentPaymentMethod,
+	purchase,
+	isDataLoading,
+	successCallback,
+} ) {
+	const currentlyAssignedPaymentMethodId = 'existingCard-' + currentPaymentMethod.stored_details_id; // TODO: make this work for paypal.
 
-	const { storedCards } = useStoredCards( wpcomGetStoredCards, false );
-	const { isStripeLoading, stripeLoadingError, stripeConfiguration, stripe } = useStripe();
+	const [ formSubmitting, setFormSubmitting ] = useState( false );
+	const translate = useTranslate();
+	const { isStripeLoading, stripeLoadingError } = useStripe();
+	const [ paymentMethods ] = useAssignablePaymentMethods();
 
-	const paypalMethod = useCreatePayPal();
+	const disabled = isStripeLoading || formSubmitting || stripeLoadingError || ! paymentMethods;
 
-	const stripeMethod = useCreateCreditCard( {
-		isStripeLoading,
-		stripeLoadingError,
-		stripeConfiguration,
-		stripe,
-		shouldUseEbanx: false,
-	} );
+	const showErrorMessage = useCallback(
+		( error ) => {
+			const message = error && error.toString ? error.toString() : error;
+			notices.error(
+				message || translate( 'An error occurred while reassigning your payment method.' )
```

Comment:
I _think_ it's safe to remove the default message here. As far as I know `showErrorMessage` will never be called without an argument, and the previous line makes sure that the argument is always somehow turned into a message.